src/wmtk/TriMesh.cpp

```diff
@@ -121,13 +129,16 @@ Tuple TriMesh::switch_tuple(const Tuple& tuple, PrimitiveType type) const
         }
         assert(lvid_new != -1);
         assert(leid_new != -1);
+
+        ConstAccessor<long> hash_accessor = get_const_cell_hash_accessor();
```

Comment:
sadly this will probably turn into `switch_tuple_slow`. Actually, perhaps it would make sense to have a small wrapper around TriMesh that holds a hashaccessor (and perhaps some of the other common ones) just to make this passing a bit easier

Reply:
We could make something like `SingleThreadMeshWrapper`.